lib/actions/dragAndDrop.js

```diff
@@ -56,14 +56,18 @@ const dragAndDropHTML5Element = async (sourcePosition, destPosition, sourceEleme
   function simulateDragAndDrop(args) {
     let sourceElement, targetElement;
     sourceElement = document.elementFromPoint(args.sourcePosition.x, args.sourcePosition.y);
-    while (sourceElement.shadowRoot) {
+    let parsedShadowRoots = [];
```

Comment:
This pattern is repeated:

```
    let parsedShadowRoots = [];
    while (sourceElement.shadowRoot && !parsedShadowRoots.includes(sourceElement.shadowRoot)) {
      parsedShadowRoots.push(sourceElement.shadowRoot);
      ...
```

Can we have this extracted to a function ?

Reply:
since these functions are executed separately in browser context we will have to pass the extracted common function as args, and those arg will be serialised as string so to again use it we will have to convert them back to function.. I feel it will make it more complex making it less readable